File: backend/app/schemas/school.py
from datetime import datetime
from pydantic import BaseModel


class SchoolQueryParams(BaseModel):
    keyword: str | None = None
    province: str | None = None
    city: str | None = None
    level: str | None = None
    school_type: str | None = None
    ownership: str | None = None
    is_985: bool | None = None
    is_211: bool | None = None
    is_double_first_class: bool | None = None
    page: int = 1
    page_size: int = 20


class SchoolListItem(BaseModel):
    id: int
    name: str
    english_name: str | None = None
    school_code: str
    province: str
    city: str
    level: str
    school_type: str
    ownership: str
    is_985: bool
    is_211: bool
    is_double_first_class: bool
    popular_majors: list | None = None

    model_config = {"from_attributes": True}


class ProvinceRead(BaseModel):
    """省份最小信息，供筛选器下拉列表。"""
    id: int
    name: str
    name_full: str
    code: str
    slug: str

    model_config = {"from_attributes": True}


class SchoolRead(SchoolListItem):
    district: str | None = None
    address: str | None = None
    longitude: float | None = None
    latitude: float | None = None
    website: str | None = None
    admission_website: str | None = None
    description: str | None = None
    created_at: datetime | None = None
    updated_at: datetime | None = None

File: frontend/src/types/school.ts
export interface School {
  id: number
  name: string
  englishName?: string
  schoolCode: string
  province: string
  city: string
  district?: string
  address?: string
  longitude?: number
  latitude?: number
  level: '本科' | '专科' | '高职'
  schoolType: '综合类' | '理工类' | '师范类' | '医药类' | '财经类' | '政法类' | '艺术类' | '体育类' | '其他'
  ownership: '公办' | '民办' | '中外合作'
  is985: boolean
  is211: boolean
  isDoubleFirstClass: boolean
  website?: string
  admissionWebsite?: string
  description?: string
  popularMajors: string[]
}

export interface ProvinceStat {
  province: string
  total: number
  undergraduateCount: number
  juniorCollegeCount: number
  count985: number
  count211: number
  doubleFirstClassCount: number
}

export interface SchoolFilter {
  keyword: string
  levels: string[]
  tags: string[]
  ownerships: string[]
  schoolTypes: string[]
  province?: string
  year?: number
  subjectType?: string
}
